src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/segmentgroup/SG01.java

```diff
@@ -0,0 +1,76 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.PerformingOrganisationNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RequesterNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ServiceProvider;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class SG01 extends SegmentGroup {
+    public static final String KEY = "S01";
+
+    // NAD+SLA?
+    @Getter(lazy = true)
+    private final Optional<PerformingOrganisationNameAndAddress> performingOrganisationNameAndAddress =
+        extractOptionalSegment(PerformingOrganisationNameAndAddress.KEY_QUALIFIER)
+            .map(PerformingOrganisationNameAndAddress::fromString);
```

Comment:
Each segment group then knows about the fields (and children groups) contained "within" it. If they are optional, the field is `Optional<T>`, if they're a required singleton it's just a normal field, and if it's any other multiplicity then a `List<T>` is needed.

In some cases for example, the document gives 0..2 for a key, but then describes 2 separate optional qualified entries for that key. In this case, each qualified entry may be listed separately.